osu.Game/Overlays/OverlayScrollContainer.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using osu.Framework.Allocation;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Effects;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics.Containers;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays
+{
+    public class OverlayScrollContainer : OsuScrollContainer
+    {
+        public ScrollToTopButton Button { get; }
+
+        private bool animaionIsRunning;
+
+        public OverlayScrollContainer()
+        {
+            AddInternal(Button = new ScrollToTopButton
+            {
+                Anchor = Anchor.BottomRight,
+                Origin = Anchor.BottomRight,
+                Margin = new MarginPadding(20),
+                Action = () =>
+                {
+                    if (animaionIsRunning)
+                        return;
+
+                    animaionIsRunning = true;
```

Comment:
Is never set back to `false`, leading to a scenario in which you could leak mouse down events to background drawables (because `OnMouseDown` yields to background drawables if this is true). See bdach/osu@7fdafef and [this video](https://streamable.com/l5qgx).

Reproduction steps:
* Scroll down to make button visible
* Click button
* Do not scroll or drag container, hold left mouse over button (underneath the scrolling container)
* Button will take the left mouse down event
* Scroll or drag container, hold left mouse over button
* Button will not take the left mouse down event